RecoEgamma/EgammaTools/plugins/EGExtraInfoModifierFromDB.cc

```diff
@@ -0,0 +1,652 @@
+  eval_ep[4] = sigma/mean/ele.trackMomentumError();
+  eval_ep[5] = tot_energy*mean/ep;
+  eval_ep[6] = tot_energy*mean/ep*sqrt(sigma/mean*sigma/mean+trkMomentumRelError*trkMomentumRelError);
+  eval_ep[7] = ele.ecalDriven();
+  eval_ep[8] = ele.trackerDrivenSeed();
+  eval_ep[9] = int(ele.classification());//eleClass;
+  eval_ep[10] = iseb;
+  
+  // CODE FOR FUTURE SEMI_PARAMETRIC
+  //double rawweight = ep_forestH_mean_[coridx]->GetResponse(eval_ep.data());
+  ////rawsigma = ep_forestH_sigma_[coridx]->GetResponse(eval.data());
+  //double weight = meanoffset + meanscale*vdt::fast_sin(rawweight);
+  ////sigma = sigmaoffset + sigmascale*vdt::fast_sin(rawsigma);
+
+  // CODE FOR STANDARD BDT
+  double weight = ep_forestH_weight_->GetResponse(eval_ep);
+  if(weight>1.) 
+    weight = 1.;
+  else if(weight<0.) 
+    weight = 0.;
+
+  double combinedMomentum = weight*ele.trackMomentumAtVtx().R() + (1.-weight)*ecor;
+  double combinedMomentumError = sqrt(weight*weight*ele.trackMomentumError()*ele.trackMomentumError() + (1.-weight)*(1.-weight)*sigmacor*sigmacor);
+
+  math::XYZTLorentzVector oldMomentum = ele.p4();
+  math::XYZTLorentzVector newMomentum = math::XYZTLorentzVector(oldMomentum.x()*combinedMomentum/oldMomentum.t(),
+								oldMomentum.y()*combinedMomentum/oldMomentum.t(),
+								oldMomentum.z()*combinedMomentum/oldMomentum.t(),
+								combinedMomentum);
+ 
+  //ele.correctEcalEnergy(combinedMomentum, combinedMomentumError);
+  ele.correctMomentum(newMomentum, ele.trackMomentumError(), combinedMomentumError);
+}
+
+void EGExtraInfoModifierFromDB::modifyObject(pat::Photon& pho) const {
+  // we encounter two cases here, either we are running AOD -> MINIAOD
+  // and the value maps are to the reducedEG object, can use original object ptr
+  // or we are running MINIAOD->MINIAOD and we need to fetch the pat objects to reference
+  edm::Ptr<reco::Candidate> ptr(pho.originalObjectRef());
+
+  if(!ph_conf.tok_photon_src.isUninitialized()) {
+    auto key = phos_by_oop.find(pho.originalObjectRef().key());
+    if( key != phos_by_oop.end() ) {
+      ptr = key->second;
+    } else {
+      throw cms::Exception("BadPhotonKey")
+        << "Original object pointer with key = " << pho.originalObjectRef().key() << " not found in cache!";
+    }
+  }
+  
+  std::array<float, 31> eval;
+  reco::SuperClusterRef sc = pho.superCluster();
+  edm::Ptr<reco::CaloCluster> theseed = sc->seed();
+  
+  // SET INPUTS
+  eval[0]  = sc->rawEnergy();
+  //eval[1]  = sc->position().Eta();
+  //eval[2]  = sc->position().Phi();
+  eval[1]  = pho.full5x5_r9();
+  eval[2]  = sc->etaWidth();
+  eval[3]  = sc->phiWidth(); 
+  const int N_ECAL = sc->clustersEnd() - sc->clustersBegin();
+  eval[4]  = std::max(0,N_ECAL - 1);
+  eval[5]  = pho.hadronicOverEm();
+  eval[6]  = rhoValue_;
+  eval[7]  = nVtx_;  
+  eval[8] = theseed->eta()-sc->position().Eta();
+  eval[9] = atan2(sin(theseed->phi()-sc->position().Phi()), cos(theseed->phi()-sc->position().Phi()));
```

Comment:
BTW, since numerical precision is a concern to get identity to what was used in training:
the eval[] values are float. The regression payloads are doubles, right?
Isn't this a problem then too?